src/Settings.js

```diff
@@ -1,20 +1,28 @@
 // @flow
+/* eslint no-console:0 */
 /**
  * This is a module for storing settings passed into KaTeX. It correctly handles
  * default settings.
  */
 
 import utils from "./utils";
+import ParseError from "./ParseError.js";
+import ParseNode from "./ParseNode";
+import {Token} from "./Token";
 
 import type { MacroMap } from "./macros";
 
+export type StrictFunction =
+    (errorCode: string, errorMsg: string, token?: Token | ParseNode) =>
+    ?(boolean | string);
```

Comment:
Exposing this level of customizability to users seems like overkill.

Reply:
This level of customizability lets users support some strict behavior and not others.  @ylemkimon at least seemed to think this would be useful, and it's not exactly hard to support.  Thoughts?

Reply:
Another point for allowing a function is that it enables custom warning functionality, if `console.warn` isn't the desired warning output.  For example, this would make our own testing easier, as `jest` doesn't really handle warnings well.

Reply:
I think it's nice to have customizability as long as we have simpler options to set them. We may add lots of features to the non-strict mode, and `strict` option may become too generic.